src/plugins/data/public/ui/query_editor/_query_editor.scss

```diff
@@ -199,24 +199,46 @@
   }
 }
 
-.suggest-widget {
-  position: relative;
+/* Fix for Monaco Editor suggestion widget width mismatch */
+.monaco-editor .suggest-widget {
+  /* We don't override the width here to maintain Monaco's default behavior */
 
-  &.visible::after {
-    position: absolute;
-    height: auto;
-    bottom: calc(-1 * ($euiFontSizeXS * 2 + $euiSizeXS) + 1px);
-    left: 0;
-    width: 100%;
-    background-color: $euiColorLightestShade;
-    border: $euiBorderThin;
-    padding: $euiSizeXS;
-    text-align: left;
-    color: $euiColorDarkShade;
-    font-size: $euiFontSizeXS;
-    content: "Tab to insert, ESC to close window";
-    display: block;
+  /* Target the suggestion list to ensure it has the same width as the widget */
+  .monaco-list {
+    width: 100% !important;
   }
+
+  /* Target the list rows to ensure they have the same width */
+  .monaco-list-row {
+    width: 100% !important;
+  }
+}
+
+/* Style for our custom status bar */
+.custom-suggest-widget-status-bar {
+  /* Base styles */
+  background-color: $euiColorLightestShade;
+  border: $euiBorderThin;
+  color: $euiColorDarkShade;
+  font-size: $euiFontSizeXS;
```

Comment:
i think all of these custom CSS variables can be changed to `$ouiX` variants safely

Reply:
Yeah would love to do that but it seems all the styles are using $euiX. I think we should clean them up together in some separate task